### PR TITLE
[enterprise-4.15] TELCODOCS-1976: Adding content for improved SR-IOV Network Operator node draining

### PR DESCRIPTION
**Pull request by amolnar-gh**
Version(s): 4.15

Issue: https://issues.redhat.com/browse/TELCODOCS-1976

Link to docs preview:

- https://79757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/hardware_networks/configuring-sriov-device.html
- https://79757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-15-release-notes.html#ocp-4-15-networking-sriov-draining

QE review:
- [x] QE has approved this change.

Additional information:
Cherrypicked from the original main and RN PRs:
* [Main](https://github.com/openshift/openshift-docs/pull/74464)
* [RN](https://github.com/openshift/openshift-docs/pull/75620)

### CODE REVIEW

**Comment by ealcaniz** [comment]
LGTM

File: modules/nw-sriov-configuring-multiple-nodes.adoc [hunk 1]
```diff
@@ -0,0 +1,171 @@
+:_mod-docs-content-type: PROCEDURE
+[id="configure-sr-iov-operator-parallel-nodes_{context}"]
+= Configuring parallel node draining during SR-IOV network policy updates
+
+By default, the SR-IOV Network Operator drains workloads from a node before every policy change.
+The Operator performs this action, one node at a time, to ensure that no workloads are affected by the reconfiguration.
```
Comment (lahinson) [inline modules/nw-sriov-configuring-multiple-nodes.adoc:6]:
```suggestion
The Operator completes this action, one node at a time, to ensure that no workloads are affected by the reconfiguration.
```

File: modules/nw-sriov-configuring-multiple-nodes.adoc [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+In large clusters, draining nodes sequentially can be time-consuming, taking hours or even days. In time-sensitive environments, you can enable parallel node draining in an `SriovNetworkPoolConfig` custom resource (CR) for faster rollouts of SR-IOV network configurations. 
+
+To configure parallel draining, use the `SriovNetworkPoolConfig` CR to create a node pool. You can then add nodes to the pool and define the maximum number of nodes in the pool that the Operator can drain in parallel. With this approach, you can enable parallel draining for faster reconfiguration while ensuring you still have enough nodes remaining in the pool to handle any running workloads.
+
+[NOTE]
+====
+A node can only belong to one SR-IOV network pool configuration. If a node is not part of a pool, it is added to a virtual, default, pool that is configured to drain one node at a time only.
```
Comment (lahinson) [inline modules/nw-sriov-configuring-multiple-nodes.adoc:14]:
```suggestion
A node can belong to only one SR-IOV network pool configuration. If a node is not part of a pool, it is added to a virtual, default pool that is configured to drain one node at a time only.
```

File: modules/nw-sriov-configuring-multiple-nodes.adoc [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+The node might restart during the draining process.
+====
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+* Install the SR-IOV Network Operator.
+* Nodes have hardware that support SR-IOV.
```
Comment (lahinson) [inline modules/nw-sriov-configuring-multiple-nodes.adoc:24]:
```suggestion
* Ensure that nodes have hardware that supports SR-IOV.
```

File: modules/nw-sriov-configuring-multiple-nodes.adoc [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  namespace: openshift-sriov-network-operator <2>
+spec:
+  maxUnavailable: 2 <3>
+  nodeSelector: <4>
+    matchLabels:
+      node-role.kubernetes.io/worker: ""
+----
+<1> Specify the name of the `SriovNetworkPoolConfig` object.
+<2> Specify namespace where the SR-IOV Network Operator is installed.
+<3> Specify an integer number, or percentage value, for nodes that can be unavailable in the pool during an update. For example, if you have 10 nodes and you set the maximum unavailable to 2, then only 2 nodes can be drained in parallel at any time, leaving 8 nodes for handling workloads.
```
Comment (lahinson) [inline modules/nw-sriov-configuring-multiple-nodes.adoc:48]:
```suggestion
<3> Specify an integer number or percentage value for nodes that can be unavailable in the pool during an update. For example, if you have 10 nodes and you set the maximum unavailable value to 2, then only 2 nodes can be drained in parallel at any time, leaving 8 nodes for handling workloads.
```

File: modules/nw-sriov-configuring-multiple-nodes.adoc [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
++
+.Example output
+[source,terminal]
+----
+NAME     AGE
+pool-1   67s <1>
+----
+<1> In this example, `pool-1` contains all the nodes with the `worker` role.
+
+To demonstrate the node draining process using the example scenario from the above procedure, complete the following steps:
```
Comment (lahinson) [inline modules/nw-sriov-configuring-multiple-nodes.adoc:139]:
```suggestion
To demonstrate the node draining process by using the example scenario from the previous procedure, complete the following steps:
```
Should this be formatted as a bulleted item like the bulleted item above it on line 124?

File: release_notes/ocp-4-15-release-notes.adoc [hunk 6]
```diff
@@ -706,6 +706,13 @@ With this release, you can use the `NodeNetworkConfigurationPolicy` resource to
 
 For example, you can configure a host network Quality of Service (QoS) policy to manage network access to host resources by an attached SR-IOV network VF. For more information, see xref:../networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.adoc#virt-example-vf-host-services_k8s_nmstate-updating-node-network-config[Node network configuration policy for virtual functions].
 
+[id="ocp-4-15-networking-sriov-draining"]
+==== Parallel node draining during SR-IOV network policy updates
+
+With this update, you can configure the SR-IOV Network Operator to drain nodes in parallel during network policy updates. The option to drain nodes in parallel enables faster rollouts of SR-IOV network configurations. You can use the `SriovNetworkPoolConfig` custom resource to configure parallel node draining and define the maximum number of nodes in the pool that the Operator can drain in parallel.
+
+For further information, see xref:../networking/hardware_networks/configuring-sriov-device.adoc#configure-sr-iov-operator-parallel-nodes_configuring-sriov-device[Configuring parallel node draining during SR-IOV network policy updates].
```
Comment (lahinson) [inline release_notes/ocp-4-15-release-notes.adoc:714]:
```suggestion
For more information, see xref:../networking/hardware_networks/configuring-sriov-device.adoc#configure-sr-iov-operator-parallel-nodes_configuring-sriov-device[Configuring parallel node draining during SR-IOV network policy updates].
```

**Review by lahinson — COMMENTED** [review]
I applied the same changes here as I did to https://github.com/openshift/openshift-docs/pull/79809.

**Pull request merged**